### PR TITLE
Esxbuilder

### PR DESCRIPTION
**Pull request by rhass**
Massive PR of stuff. I can rework these commits if its too much.

### CODE REVIEW

File: bin/buildkite.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+# Environment Varibles which need to be set in the job config
+BUILDKITE_ACCESS_TOKEN = ENV['BUILDKITE_ACCESS_TOKEN']
+BUILDKITE_PROJECT = ENV['BUILDKITE_PROJECT'] ||= 'vsphere-baker-windows'
+# Environment varibles defined by BuildKite automagically.
+BUILDKITE_ORGANIZATION = ENV['BUILDKITE_ORGANIZATION_SLUG'] ||= 'chef'
+BUILDKITE_BRANCH = ENV['BUILDKITE_BRANCH'] ||= 'master'
+# Space delimited list of Packer JSON files to build regardless of changes.
+FORCE_BUILD_LIST= ENV['FORCE_BUILD_LIST'] || ''
+IGNORED_FILES = %w(
+  gitignore
```
Comment (sersut) [inline bin/buildkite.rb]: `.gitignore` ?

File: bin/buildkite.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  http.use_ssl = true
+
+  request ||= Net::HTTP::Get.new(uri.request_uri)
+  @response ||= http.request(request)
+
+  @logger.debug(@response)
+  @logger.debug(@response.code)
+  @logger.debug(@response.body)
+
+  if @response.code.to_i >= 400
```
Comment (sersut) [inline bin/buildkite.rb]: `@response.value` is the equivalent of this:

https://github.com/chef/mixlib-install/blob/master/lib/mixlib/install/backend/omnitruck.rb#L68-L69

File: bin/buildkite.rb [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+buildlist.collect! { |b| b.gsub!('.json', '') }
+
+if buildlist.empty?
+  @logger.info("No template changes found to build.")
+else
+  # We need to do a more complex threading implementation to be able to build
+  # these in parallel as Packer has a bug in which it fails when an output directory
+  # already exists on the system. Attempting to use make to parallelize this will
+  # result in the same
+  buildlist.each do |template|
```
Comment (sersut) [inline bin/buildkite.rb:123]: See the parallelization in chef-acceptance [here](https://github.com/chef/chef-acceptance/blob/ce41489f4bc1fb58bca14146d2a713ff10e2bac0/lib/chef-acceptance/application.rb#L50-L54) as an example. That might be useful here.

**Comment by schisamo** [comment]
Is any of the above squashable?

**Comment by rhass** [comment]
I will try to squish some of this down today.

**Comment by rhass** [comment]
Squishing hasn't quite worked out too well so far. I'll try some other things....

**Comment by yzl** [comment]
👍  pending squash.

File: floppy/upgrade-wmf.bat [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+
+:WMF4
+reg query "HKLM\SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine" /v PowerShellVersion 2>&1 | find "4.0" 2>&1>NUL
+if %ERRORLEVEL% equ 0 (
+    echo Windows Management Framework 4.0 is already installed
+    goto EXIT
+)
+
+echo Downloaing Windows Management Framwork 4.0 (%BITNESS%)
```
Comment (scotthain) [inline floppy/upgrade-wmf.bat:61]: `Downloading`

**Comment by scotthain** [comment]
Other than very minor spelling issue 👍

**Comment by rhass** [comment]
Squashed out the useless stuff.

**Pull request merged**